middleware/proxy/google.go

```diff
@@ -218,23 +219,30 @@ func newUpstream(hosts []string, old *staticUpstream) Upstream {
 			Conns:       0,
 			Fails:       0,
 			FailTimeout: upstream.FailTimeout,
-			Unhealthy:   false,
+			OkUntil:     time.Now().Add(time.Duration(24 * 365 * 100 * time.Hour)), // now + 100 years
```

Comment:
but you can special case `time.Time{}` in the code does that check as well, no? Yes, that makes that check somewhat more ugly, because you need to check for the special value. OTOH this 100 years is also a legal time vaue. I slightly lean into make an `IsZero` time the special thing here.